From 68dce680d5f5afe9074689bdd34bfbcfe0d121d9 Mon Sep 17 00:00:00 2001
From: Ben Peled <bpeled@marvell.com>
Date: Thu, 20 May 2021 08:12:39 +0300
Subject: [PATCH 1567/1921] crypto: inside-secure: store cell-index value

This patch stores the cell-index which is read from
the device-tree.

Change-Id: I61b1d9862801d1564599e3d79133594a51fcb265
Signed-off-by: Igal Liberman <igall@marvell.com>
Signed-off-by: Ben Peled <bpeled@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51186
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/crypto/inside-secure/safexcel.c | 6 ++++++
 drivers/crypto/inside-secure/safexcel.h | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/drivers/crypto/inside-secure/safexcel.c b/drivers/crypto/inside-secure/safexcel.c
index e83a52184ebb..2d910f4eafe6 100644
--- a/drivers/crypto/inside-secure/safexcel.c
+++ b/drivers/crypto/inside-secure/safexcel.c
@@ -1579,6 +1579,12 @@ static int safexcel_probe(struct platform_device *pdev)
 		return PTR_ERR(priv->base);
 	}
 
+	ret = of_property_read_u32(dev->of_node, "cell-index", &priv->id);
+	if (ret) {
+		dev_err(dev, "failed to read cell-index property\n");
+		return ret;
+	}
+
 	priv->clk = devm_clk_get(&pdev->dev, NULL);
 	ret = PTR_ERR_OR_ZERO(priv->clk);
 	/* The clock isn't mandatory */
diff --git a/drivers/crypto/inside-secure/safexcel.h b/drivers/crypto/inside-secure/safexcel.h
index 1a6b39154477..dc8eb6e80efe 100644
--- a/drivers/crypto/inside-secure/safexcel.h
+++ b/drivers/crypto/inside-secure/safexcel.h
@@ -729,6 +729,8 @@ struct safexcel_crypto_priv {
 	atomic_t ring_used;
 
 	struct safexcel_ring *ring;
+
+	int id;
 };
 
 struct safexcel_context {
-- 
2.31.1

